src/features/calendar/components/CalendarEventFilter/EventFilterPane.tsx

```diff
@@ -0,0 +1,176 @@
+          reset={disableReset}
+          userText={state.text}
+        />
+        <Box display="flex" flexDirection="column" sx={{ mt: 2 }}>
+          <CheckboxFilterList
+            filterCategory={'selectedActions'}
+            onClickAll={(filterCategory) => {
+              store.dispatch(
+                filterUpdated({
+                  filterCategory,
+                  selectedFilterValue: Object.values(ACTION_FILTER_OPTIONS),
+                })
+              );
+            }}
+            onClickCheckbox={(e, value) => handleCheckBox(e, value)}
+            onClickNone={(filterCategory) => handleSelectNone(filterCategory)}
+            options={Object.values(ACTION_FILTER_OPTIONS).map((value) => ({
+              label: messages.eventFilter.filterOptions.actionFilters[value](),
+              value,
+            }))}
+            state={state.selectedActions}
+            title={messages.eventFilter.filterOptions.actionFilters.title()}
+          />
+          <CheckboxFilterList
+            filterCategory={'selectedStates'}
+            onClickAll={(filterCategory) => {
+              store.dispatch(
+                filterUpdated({
+                  filterCategory,
+                  selectedFilterValue: Object.values(STATE_FILTER_OPTIONS),
+                })
+              );
+            }}
+            onClickCheckbox={(e, value) => handleCheckBox(e, value)}
+            onClickNone={(filterCategory) => handleSelectNone(filterCategory)}
+            options={Object.values(STATE_FILTER_OPTIONS).map((value) => ({
+              label: messages.eventFilter.filterOptions.stateFilters[value](),
+              value,
+            }))}
+            state={state.selectedStates}
+            title={messages.eventFilter.filterOptions.stateFilters.title()}
+          />
+          <ZUIFuture future={typesModel.getTypes()}>
+            {(data) => {
+              return (
+                <CheckboxFilterList
+                  filterCategory={'selectedTypes'}
+                  onClickAll={(filterCategory) => {
+                    store.dispatch(
+                      filterUpdated({
+                        filterCategory,
+                        selectedFilterValue: data.map(
+                          (value: ZetkinActivity) => `${value.id}`
+                        ),
+                      })
+                    );
+                  }}
+                  onClickCheckbox={(e, value) => handleCheckBox(e, value)}
+                  onClickNone={(filterCategory) =>
+                    handleSelectNone(filterCategory)
+                  }
+                  options={data
+                    .map((value: ZetkinActivity) => ({
+                      label: value.title,
+                      value: `${value.id}`,
+                    }))
+                    .sort((a, b) =>
+                      a.label.toLowerCase().localeCompare(b.label.toLowerCase())
+                    )}
```

Comment:
Great that you're sorting the list here, so that there doesn't need to be any logic for sorting (in some cases only) inside `CheckboxFilterList`!